hrjobcontract/js/src/job-contract/controllers/contract.js

```diff
@@ -69,8 +69,10 @@ define([
         _.extend($scope.health, newScope.health || contractRevisionIdObj);
         _.extend($scope.pension, newScope.pension || contractRevisionIdObj);
 
-        _.each($scope.leave, function (leaveType, leaveTypeId) {
-          _.extend(leaveType, newScope.leave ? newScope.leave[leaveTypeId] || contractRevisionIdObj : contractRevisionIdObj);
+        _.each($scope.leave, function (leaveType) {
```

Comment:
put this in a function and document it, cause it's hard to understand what's going on

Reply:
done.